src/middlewares/roleCheck.ts

```diff
@@ -0,0 +1,37 @@
+import { NextFunction, Request, Response } from "express";
+import prisma from "../prisma";
+
+export default function roleCheck(roles: number[]) {
+    return async (req: Request, res: Response, next: NextFunction) => {
+        const userId = res.locals.jwtPayload.userId;
+        const courseId = parseInt(req.params.course_id);
+        const user = await prisma.enrollment.findFirstOrThrow({
```

Comment:
Tænker vi skal bruge en `findUniqueOrThrow`, og husk en `.catch` efter? evt. i en `then` så du bare kan tage rollen ud istedet for et `user` object